src/planners/bidirectional.jl

```diff
@@ -0,0 +1,130 @@
+    forward::F
+    backward::B
+end
+
+BiDirectionalPlanner(;kwargs...) = BiDirectionalPlanner(
+    	ForwardPlanner(; kwargs...),
+    	BackwardPlanner(; kwargs...)
+    )
+
+"Bidirectional  Greedy best-first search, with cycle checking."
+BiGreedyPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...)
+    )
+
+
+"Bidirectional A* search."
+BiAStarPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, kwargs...)
+    )
+
+function solve(planner::BiDirectionalPlanner,
+               domain::Domain, state::State, spec::Specification)
+
+    # Simplify goal specification
+    forward_spec = simplify_goal(spec, domain, state)
+    backward_spec = BackwardSearchGoal(spec, state)
+    # Precompute heuristic information
+    precompute!(planner.forward.heuristic, domain, state, forward_spec)
+    precompute!(planner.backward.heuristic, domain, state, backward_spec)
+    
+	forward_search_tree, forward_queue = init_forward(planner.forward, domain, state, spec)
+	backward_search_tree, backward_queue = init_backward(planner.backward, domain, state, spec)
+
+    status, node_id, count = search!(planner, domain, 
+    	forward_spec, forward_search_tree, forward_queue,
+    	backward_spec, backward_search_tree, backward_queue)
+
+    # Reconstruct plan and return solution
+    if status != :failure
+        forward_plan, forward_traj = reconstruct(node_id, forward_search_tree)
+        backward_plan, backward_traj = reconstruct(node_id, backward_search_tree)
+        reverse!(backward_plan); reverse!(backward_traj)
+        plan = vcat(forward_plan, backward_plan)
+        traj = vcat(forward_traj, backward_traj)
+        if planner.forward.save_search || planner.backward.save_search
+            return BiPathSearchSolution(status, plan, traj,
+                                      count, forward_search_tree, forward_queue, 
+                                      backward_search_tree, backward_queue)
+        else
+            return BiPathSearchSolution(status, plan, traj)
+        end
+    elseif planner.forward.save_search || planner.backward.save_search
+        S = typeof(state)
+        return BiPathSearchSolution(status, Term[], S[],
+                                  count, search_tree, queue)
+    else
+        return NullSolution(status)
+    end
+end
+
+function search!(planner::BiDirectionalPlanner, domain::Domain, 
+                 forward_spec::Specification, forward_search_tree::Dict{UInt,<:PathNode}, forward_queue::PriorityQueue,
+                 backward_spec::Specification, backward_search_tree::Dict{UInt,<:PathNode}, backward_queue::PriorityQueue)
+	max_nodes = max(planner.forward.max_nodes)
+	max_time = max(planner.forward.max_time)
```

Comment:
Rather than using the forward planners parameters, I think maybe it'd be a good idea to store this separately in the `BiDirectionalPlanner` struct? Also, I'm a little confused by why you're using the `max` function here.